### ISSUE TITLE
Add RealMLP model family via pytabkit

### ISSUE THREAD

**Issue by haukesteffen** [haukesteffen/TabularShenanigans#231]
## Context

The current model registry is all tree-based or linear models. For ensemble diversity in blends, a well-tuned MLP that produces structurally different predictions is more valuable than another GBM variant. RealMLP (from the "Tuned MLPs are Competitive" line of work) is the most practical deep learning addition: `pytabkit` provides sklearn-compatible `RealMLP_TD_Classifier` / `RealMLP_TD_Regressor` with strong defaults out of the box, and it requires no pipeline rework.

Unlike TabNet or FT-Transformer, RealMLP has no dataset size limits and minimal hyperparameter sensitivity. The primary value is low-correlation OOF predictions for blending — it won't individually beat tuned GBMs on Playground Series-scale data, but it fills a different region of prediction space.

### 1. Dependency in `pyproject.toml`

Add `pytabkit` as an optional dependency under a new `neural` extra (separate from `boosters` since it pulls in PyTorch):

```toml
neural = [
    "pytabkit>=1.0.0",
]
```

### 2. Model builders in `models.py`

Add builder functions following the existing lazy-import pattern (see `_build_lightgbm_regressor` for reference):

| Builder | Estimator class |
|---|---|
| `_build_realmlp_regressor` | `pytabkit.RealMLP_TD_Regressor` |
| `_build_realmlp_classifier` | `pytabkit.RealMLP_TD_Classifier` |

Builder details:
- Lazy-import `pytabkit` with a clear `ImportError` message pointing to `uv sync --extra neural`
- Pass `random_state` to the estimator's seed/random_state parameter
- Use pytabkit's built-in defaults — the whole point of RealMLP is that the defaults are well-tuned
- Merge `parameter_overrides` via `_merge_model_params` as usual
- RealMLP uses PyTorch internally for GPU — set `device='cuda'` when `resolved_compute_target == 'gpu'`, otherwise `device='cpu'`
- No `BinaryLabelEncodingClassifier` wrapper needed — pytabkit's classifier is sklearn-compatible with `predict_proba` and `classes_`
- Verify this during implementation; wrap if needed

### 3. Optuna tuning space in `models.py`

Define a tuning space. Exact parameter names should be verified against the installed `pytabkit` version, but the likely candidates are:

| Parameter | Range |
|---|---|
| `n_epochs` | [50, 500] |
| `batch_size` | {128, 256, 512} |
| `lr` | [1e-4, 1e-2, log] |
| `hidden_sizes` | {[128], [256], [128, 128], [256, 128]} |
| `dropout_p` | [0.0, 0.5] |
| `weight_decay` | [1e-6, 1e-2, log] |

### 4. Model registry entries in `models.py`

Register in `MODEL_REGISTRY` for both `regression` and `binary`:

```python
"realmlp": ModelDefinition(
    model_id="realmlp",
    model_name="RealMLP_TD_Regressor",  # or Classifier
    builder=_build_realmlp_regressor,    # or classifier
    tuning_space_builder=_build_realmlp_tuning_space,
    supports_sparse_preprocessed_input=False,
    supports_gpu_native_dense_onehot_input=False,
),
```

- `supports_sparse=False`: pytabkit expects dense numpy input
- `supports_gpu_native_dense_onehot=False`: not a cuML model
- No `fit_kwargs_builder`: no special fit kwargs like CatBoost's `cat_features`
- No `supports_native_categorical_preprocessing`: uses standard preprocessed input

### 5. No GPU execution routing

RealMLP is **not** a cuML model — it uses PyTorch for GPU internally. No entries needed in `execution_routing.py`. The `device` parameter in the builder handles GPU dispatch directly, independent of the cuML routing system.

### 6. Sweep candidate generation in `scripts/generate_sweep_candidates.py`

Add `realmlp` to `CPU_MODELS` (since it has no cuML GPU path, even though it uses PyTorch GPU internally). It supports all standard categorical preprocessors (onehot/ordinal/frequency) via dense input.

Alternatively, add a new group `NEURAL_MODELS` if the distinction matters for sweep organization.

### What doesn't change

Preprocessing, blend, evaluation, MLflow logging, config validation, and `validate_model_preprocessing_compatibility` all work as-is. The `resolve_model_matrix_output_kind` function will return `"dense_array"` by default since `supports_sparse=False`.

### Design note: preprocessing

RealMLP internally normalizes numeric features and embeds categoricals when given raw input. In this pipeline it receives already-preprocessed dense arrays, so its internal normalization acts on top of ours. This is fine — `standardize` candidates may slightly outperform `median` since the double-normalization is a no-op, but all preprocessor combinations are valid.

## Acceptance criteria

- [ ] `pytabkit` added as optional `neural` extra in `pyproject.toml`
- [ ] Builder functions for regressor and classifier with lazy import and clear error message
- [ ] Builders pass `device='cuda'` on GPU, `device='cpu'` on CPU
- [ ] Default parameters from pytabkit used (not overridden without reason)
- [ ] Tuning space defined with parameter names verified against installed pytabkit
- [ ] `MODEL_REGISTRY` entries for both `regression` and `binary` with correct flags
- [ ] `generate_sweep_candidates.py` includes `realmlp` candidates
- [ ] Smoke test: single `realmlp` candidate trains successfully on s6e3 (both CPU and, if available, GPU)
